memorystore/snippets/connect_to_db_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/longrunning/autogen/longrunningpb"
+	memorystore "cloud.google.com/go/redis/apiv1"
+	redispb "cloud.google.com/go/redis/apiv1/redispb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"github.com/google/uuid"
+)
+
+var instanceID string
+
+func TestMain(m *testing.M) {
+
+	// Set up
+	tc, ok := testutil.ContextMain(m)
+	if !ok {
+		log.Fatal("test project not set up properly")
+		return
+	}
+	parent := fmt.Sprintf("projects/%s/locations/us-central1/", tc.ProjectID)
+
+	ctx := context.Background()
+	id := uuid.New()
+	instanceID = fmt.Sprintf("test-instance-%s", id.String())
+
+	adminClient, err := memorystore.NewCloudRedisClient(ctx)
+	if err != nil {
+		log.Fatal("can't instantiate MemoryStore Redis admin client")
+		return
+	}
+
+	req := &redispb.CreateInstanceRequest{
+		Parent:     parent,
+		InstanceId: instanceID,
+		Instance: &redispb.Instance{
+			Name: fmt.Sprintf("%s/%s", parent, instanceID),
+		},
+	}
+
+	op, err := adminClient.CreateInstance(ctx, req)
+	if err != nil {
+		return
+	}
+
+	for {
```

Comment:
I think this can be simplified with the `WaitWithInterval` method: https://pkg.go.dev/cloud.google.com/go/longrunning#Operation.WaitWithInterval

(IIRC, getting operation status uses quota, so the interval solution will be more reliable)

Reply:
Good call. The `CreateInstanceOperation` type exposes a `Wait()` method, so I'll use that.